### PR TITLE
[BinaryFormat][LoongArch] Define psABI v2.30 relocs

### PR DESCRIPTION
**Pull request by SixWeining**

### CODE REVIEW

**Comment by SixWeining** [comment]
Add @xen0n @xry111 @MQ-mengqing

**Review by xen0n — APPROVED** [review]

**Review by MQ-mengqing — APPROVED** [review]

**Review by wangleiat — APPROVED** [review]
LGTM

**Review by MaskRay — APPROVED** [review]

File: llvm/unittests/Object/ELFTest.cpp [hunk 1]
```diff
@@ -77,6 +77,10 @@ TEST(ELFTest, getELFRelocationTypeNameForLoongArch) {
             getELFRelocationTypeName(EM_LOONGARCH, R_LARCH_TLS_TPREL64));
   EXPECT_EQ("R_LARCH_IRELATIVE",
             getELFRelocationTypeName(EM_LOONGARCH, R_LARCH_IRELATIVE));
+  EXPECT_EQ("R_LARCH_TLS_DESC32",
```
Comment (MaskRay) [inline llvm/unittests/Object/ELFTest.cpp:80]: You don't need to update this unittest. Currently VE/LoongArch add relocation types here but the llvm-readobj test suffices.

Comment (SixWeining) [reply llvm/unittests/Object/ELFTest.cpp]: Removed. Thanks.

**Pull request merged**